### PR TITLE
Rotate service accout keys in secret manager

### PR DESCRIPTION
**Pull request by bipinupd**
Rotate service account keys store in secret manager.

### CODE REVIEW

File: examples/rotate-service-account-keys/main.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        project_id = json_key['project_id']
+        service_account = json_key['client_email']
+        key_id = json_key['private_key_id']
+
+        service = googleapiclient.discovery.build('iam', 'v1')
+
+        ### Create a new service account and add a key as a new secret version
+        new_key = service.projects().serviceAccounts().keys().create(
+            name='projects/' + project_id + '/serviceAccounts/' +
+            service_account,
```
Comment (danieldeleo) [inline examples/rotate-service-account-keys/main.py:46]:
```suggestion
            name=f'projects/{project_id}/serviceAccounts/{service_account}',
```

File: examples/rotate-service-account-keys/main.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            body={}).execute()
+        new_service_key_json = base64.b64decode(new_key['privateKeyData'])
+        client.add_secret_version(parent=secret_name,
+                                  payload={'data': new_service_key_json})
+
+        ### Delete the old service account
+        key_to_delete = "projects/{project_id}/serviceAccounts/{service_account}/keys/{key_id}".format(
+            project_id=project_id,
+            service_account=service_account,
+            key_id=key_id)
```
Comment (danieldeleo) [inline examples/rotate-service-account-keys/main.py:56]:
```suggestion
        key_to_delete = f'projects/{project_id}/serviceAccounts/{service_account}/keys/{key_id}'
```

File: examples/rotate-service-account-keys/main.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import base64
+import json
+import googleapiclient.discovery
+from google.cloud import secretmanager
+
+
+def rotate_service_account_keys_in_secret_manager(event, context):
+
+    event_type = event["attributes"]["eventType"]
+    secret_name = event["attributes"]["secretId"]
```
Comment (danieldeleo) [inline examples/rotate-service-account-keys/main.py:24]:
```suggestion
    event_type = event['attributes']['eventType']
    secret_name = event['attributes']['secretId']
```

File: examples/rotate-service-account-keys/main.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    if (event_type == "SECRET_ROTATE"):
```
Comment (danieldeleo) [inline examples/rotate-service-account-keys/main.py:26]:
```suggestion
    if (event_type == 'SECRET_ROTATE'):
```

File: examples/rotate-service-account-keys/main.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        key_to_delete = "projects/{project_id}/serviceAccounts/{service_account}/keys/{key_id}".format(
+            project_id=project_id,
+            service_account=service_account,
+            key_id=key_id)
+        service.projects().serviceAccounts().keys().delete(
+            name=key_to_delete).execute()
+
+        ### Disable and delete the old secret version
+        client.disable_secret_version(
+            request={"name": current_secret_with_version})
```
Comment (danieldeleo) [inline examples/rotate-service-account-keys/main.py:62]:
```suggestion
            request={'name': current_secret_with_version})
```

File: examples/rotate-service-account-keys/main.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        client.destroy_secret_version(
+            request={"name": current_secret_with_version})
```
Comment (danieldeleo) [inline examples/rotate-service-account-keys/main.py:64]:
```suggestion
            request={'name': current_secret_with_version})
```

File: examples/rotate-service-account-keys/Readme.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+   Now add a role binding, the above command with print the service account to use.
+
+    ```
+    gcloud pubsub topics add-iam-policy-binding PUBSUB_TOPIC_NAME \
+    --member "serviceAccount:service-PROJECT_NUMBER@gcp-sa-secretmanager.iam.gserviceaccount.com" \
+    --role "roles/pubsub.publisher"
+    ```
+
+- **Step 4:** Deploy the function.
```
Comment (boredabdel) [inline examples/rotate-service-account-keys/Readme.md:46]: You may want to add a step to configure the Cloud Function Service Account to be able to Create Service Account Keys and push them to SM.

**Comment by boredabdel** [comment]
Also the linter is failing. Please follow the contribution guide (https://github.com/GoogleCloudPlatform/professional-services/blob/main/CONTRIBUTING.md) to run the LInter locally and amend you changes.

File: examples/rotate-service-account-keys/Readme.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+   Now add a role binding, the above command with print the service account to use.
+
+    ```
+    gcloud pubsub topics add-iam-policy-binding PUBSUB_TOPIC_NAME \
+    --member "serviceAccount:service-PROJECT_NUMBER@gcp-sa-secretmanager.iam.gserviceaccount.com" \
+    --role "roles/pubsub.publisher"
+    ```
+
+- **Step 5:** Deploy the function.
```
Comment (boredabdel) [inline examples/rotate-service-account-keys/Readme.md:46]: Sorry for being too insistent but did you meant this to be Step4 ?

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**